### ISSUE TITLE
Identifiers with backticks are split by special characters in hover

### ISSUE THREAD

**Issue by kubukoz** [scalameta/metals-vscode#1037]
### Describe the bug

Identifiers that are defined using the backtick syntax get weird treatment when being highlighted on hover.

To reproduce

1. Write
 
```scala
oject demo {
  val `Example+Value` = 42

  println(`Example+Value`)
}
```

2. Hold command/ctrl and hover on `Example` in the println

### Expected behavior

The whole "`Example+Value`" (backticks included) is highlighted

### Operating system

macOS

### Editor/Extension

VS Code

### Version of Metals

0.11.6+67-926ec9a3-SNAPSHOT

**Comment by ckipp01** [comment]
> Yeah, this might be an issue in VS Code

I have the same issue in `nvim-metals`. I more meant that we return:
```
"```scala\nval Example+Value: Int\n```"
```
But I wonder in these situations if we should return:

```
"```scala\nval `Example+Value`: Int\n```"
```

**Comment by tgodzik** [comment]
Yeah, the color could be ok in that case, but we still get the wrong part highlighted on command + click. Hover is a separate issue.

**Comment by harpocrates** [comment]
Do you know why VS code is underlining only a subset of a backticked identifier on command+hover? Is there some sort of lexer inside VS code that would need to be fixed or is this something that could be handled inside of `metals-vscode`?

<img width="228" alt="Screenshot 2023-11-01 at 3 06 08 PM" src="https://github.com/scalameta/metals-vscode/assets/10766081/34985ec8-faf5-4650-816c-de063c7f7eca">

**Comment by tgodzik** [comment]
Looks like that one is handled by VS Code. I don't see any data being sent by LSP that would span that smaller range. Unless VS Code is working of https://github.com/scala/vscode-scala-syntax that we use for basic syntax highlight :thinking: but backticks should be handled there
